packages/labs/cli/src/test/temp-filesystem-rig.ts

```diff
@@ -0,0 +1,196 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as fs from 'fs/promises';
+import * as pathlib from 'path';
+import {fileURLToPath} from 'url';
+
+import type {Stats} from 'fs';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = pathlib.dirname(__filename);
+const repoRoot = pathlib.resolve(__dirname, '..', '..', '..', '..');
+
+/**
+ * A test rig for managing a temporary filesystem.
+ */
+export class FilesystemTestRig {
```

Comment:
We should move something like this into the internal test package.

Reply:
+1, this was extracted from wireit

Reply:
Added in https://github.com/lit/lit/pull/2875